src/scripts/main.js

```diff
@@ -1,3 +1,331 @@
+
+  game.matrix.forEach(col => {
+    const tr = document.createElement('tr');
+
+    tr.className = 'field-row';
+
+    col.forEach(row => {
+      const td = document.createElement('td');
+
+      if (row !== 0) {
+        td.className = `field-cell field-cell--${row}`;
+
+        td.textContent = row;
+      } else if (row === 0) {
+        td.className = `field-cell`;
+      }
+
+      tr.append(td);
+    });
+
+    gameFieldBody.append(tr);
+  });
+
+  renderMessage('message-start', 'Press "Start" to begin game. Good luck!');
+
+  const isWin = isGameWin();
+
+  if (!isWin) {
+    isGameLose();
+  }
+}
+
+function addRandomCell() {
+  const emptyCell = [];
+
+  game.matrix.forEach((col, colInd) => {
+    col.forEach((row, rowInd) => {
+      if (row === 0) {
+        emptyCell.push([colInd, rowInd]);
+      }
+    });
+  });
+
+  const randomCellNum = Math.random() <= 0.1 ? 4 : 2;
+  const randomPosition = Math.floor(Math.random() * emptyCell.length);
+
+  const randomCol = emptyCell[randomPosition][0];
+  const randomRow = emptyCell[randomPosition][1];
+
+  game.matrix[randomCol][randomRow] = randomCellNum;
+}
+
+function checkEmptyTail(col, row, currCol, currRow) {
+  if (currCol !== col || currRow !== row) {
+    game.matrix[currCol][currRow] = game.matrix[col][row];
+    game.matrix[col][row] = 0;
+
+    game.moved = true;
+  }
+}
+
+function searchEmptyTails(move) {
+  if (move === 'up') {
+    for (let col = 1; col < 4; col++) {
+      for (let row = 0; row < 4; row++) {
+        if (game.matrix[col][row] === 0) {
+          continue;
+        }
+
+        let currCol = col;
+
+        while (currCol > 0 && game.matrix[currCol - 1][row] === 0) {
+          currCol--;
+        }
+
+        checkEmptyTail(col, row, currCol, row);
+      }
+    }
+  } else if (move === 'down') {
+    for (let col = 2; col >= 0; col--) {
+      for (let row = 0; row < 4; row++) {
+        if (game.matrix[col][row] === 0) {
+          continue;
+        }
+
+        let currCol = col;
+
+        while (currCol < 3 && game.matrix[currCol + 1][row] === 0) {
+          currCol++;
+        }
+
+        checkEmptyTail(col, row, currCol, row);
+      }
+    }
+  } else if (move === 'left') {
+    for (let col = 0; col < 4; col++) {
+      for (let row = 1; row < 4; row++) {
+        if (game.matrix[col][row] === 0) {
+          continue;
+        }
+
+        let currRow = row;
+
+        while (currRow > 0 && game.matrix[col][currRow - 1] === 0) {
+          currRow--;
+        }
+
+        checkEmptyTail(col, row, col, currRow);
+      }
+    }
+  } else if (move === 'right') {
+    for (let col = 0; col < 4; col++) {
+      for (let row = 2; row >= 0; row--) {
+        if (game.matrix[col][row] === 0) {
+          continue;
+        }
+
+        let currRow = row;
+
+        while (currRow < 3 && game.matrix[col][currRow + 1] === 0) {
+          currRow++;
+        }
+
+        checkEmptyTail(col, row, col, currRow);
+      }
+    }
+  }
```

Comment:
too much duplication here. think about moving this logic to a function. also, you can combine this logic with `checkSameTail`